mlflow/pytorch/__init__.py

```diff
@@ -556,14 +556,16 @@ class LinearNNModel(torch.nn.Module):
     mlflow_model.save(os.path.join(path, MLMODEL_FILE_NAME))
 
     if conda_env is None:
-        default_reqs = get_default_pip_requirements()
         if pip_requirements is None:
+            default_reqs = get_default_pip_requirements()
             # To ensure `_load_pyfunc` can successfully load the model during the dependency
             # inference, `mlflow_model.save` must be called beforehand to save an MLmodel file.
             inferred_reqs = mlflow.models.infer_pip_requirements(
                 model_data_path, FLAVOR_NAME, fallback=default_reqs,
             )
             default_reqs = sorted(set(inferred_reqs).union(default_reqs))
+        else:
+            default_reqs = None
```

Comment:
The main reason I'd like to introduce this change is pytorch flavor.

```python
# This user wants to disable auto dependency inference by specifying `pip_requirements`
import mlflow
import torch

model = torch.nn.Linear(4, 1)
model.eval()

with mlflow.start_run():
    mlflow.pytorch.log_model(
        model,
        artifact_path="model",
        pip_requirements=mlflow.pytorch.get_default_pip_requirements()
    )
```

This code shows two duplicate warning messages:

```
2021/08/17 19:45:37 WARNING mlflow.utils.requirements_utils: Found torch version (1.9.0+cpu) contains a local version label (+cpu). MLflow logged a pip requirement for this package as 'torch==1.9.0' without the local version label to make it installable from PyPI. To specify pip requirements containing local version labels, please use `conda_env` or `pip_requirements`.
2021/08/17 19:45:37 WARNING mlflow.utils.requirements_utils: Found torch version (1.9.0+cpu) contains a local version label (+cpu). MLflow logged a pip requirement for this package as 'torch==1.9.0' without the local version label to make it installable from PyPI. To specify pip requirements containing local version labels, please use `conda_env` or `pip_requirements`.
```